tests/unit/adapters/osf-adapter-test.js

```diff
@@ -1,12 +1,401 @@
-import { moduleFor, test } from 'ember-qunit';
+import Ember from 'ember';
+import {
+    moduleFor
+} from 'ember-qunit';
+import test from 'dummy/tests/ember-sinon-qunit/test';
+import FactoryGuy, {
+    manualSetup
+} from 'ember-data-factory-guy';
+
+import DS from 'ember-data';
+import OsfAdapter from 'ember-osf/adapters/osf-adapter';
 
 moduleFor('adapter:osf-adapter', 'Unit | Adapter | osf adapter', {
-  // Specify the other units that are required for this test.
-  // needs: ['serializer:foo']
+    needs: [
+        'model:user', 'model:node', 'model:institution', 'model:registration', 'model:log', 'model:comment', 'model:contributor', 'model:file-provider', 'model:node-link', 'model:draft-registration',
+        'adapter:osf-adapter', 'adapter:node', 'adapter:user',
+        'serializer:node',
+        'service:session',
+        'transform:links', 'transform:embed'
+    ],
+    beforeEach() {
+        manualSetup(this.container);
+    }
+});
+
+test('#buildURL appends a trailing slash if missing', function(assert) {
+    var url = 'https://api.osf.io/v2/users/me';
+    this.stub(
+        DS.JSONAPIAdapter.prototype,
+        'buildURL',
+        function() {
+            return url;
+        }
+    );
+    let adapter = this.subject();
+    let user = FactoryGuy.make('user');
+    let result = adapter.buildURL(
+        'user',
+        'me',
+        user._internalModel.createSnapshot(),
+        'findRecord'
+    );
+    assert.notEqual(url, result);
+    assert.equal(result.slice(-1), '/');
+});
+
+test('#buildURL uses relationship links if available for delete, update, and find', function(assert) {
+    let url = 'https://api.osf.io/v2/users/me/rel/';
+    let adapter = this.subject();
+    let user = FactoryGuy.make('user', {
+        links: {
+            self: url
+        }
+    });
+    ['delete', 'update', 'find'].forEach(verb => {
+        let result = adapter.buildURL(
+            'user',
+            'me',
+            user._internalModel.createSnapshot(),
+            `${verb}Record`
+        );
+        assert.equal(url, result);
+    });
+});
+
+test('#buildURL uses snapshot.adapterOptions.url if available', function(assert) {
```

Comment:
a) That the option is included, and,
b) That it does whatever it's supposed to do

From an initial read, I wasn't 100% clear on the functionality this is intended to provide, and it would be good to have an authoritative source on what and how this should work.